### PR TITLE
only use ingresses in the same namespace for dns records

### PR DESCRIPTION
**Pull request by philbrookes**
only use ingresses in the same namespace for dns records

### CODE REVIEW

File: pkg/reconciler/ingress/ingress.go [hunk 1]
```diff
@@ -320,7 +320,7 @@ func (c *Controller) reconcileLeaf(ctx context.Context, rootName string, ingress
 	if err != nil {
 		return err
 	}
-	others, err := c.lister.List(sel)
+	others, err := c.lister.Ingresses(ingress.Namespace).List(sel)
```
Comment (mikenairn) [inline pkg/reconciler/ingress/ingress.go:323]: I think this is probably right, but why did the selector not prevent this from picking up on ingresses not owned by this one?

Comment (philbrookes) [reply pkg/reconciler/ingress/ingress.go:323]: because the owner ingress has the same name in both namespaces.

Comment (mikenairn) [reply pkg/reconciler/ingress/ingress.go:323]: Ah, it just uses name `fmt.Sprintf("%s=%s", ownedByLabel, ingress.Name)`. Should all the other places we use  ` c.lister.List(sel)` be updated as well?

Comment (philbrookes) [reply pkg/reconciler/ingress/ingress.go:323]: hmm yeah good shout. I'll do that and retest

Comment (philbrookes) [reply pkg/reconciler/ingress/ingress.go:323]: done this now.

**Review by astefanutti — APPROVED** [review]

**Pull request merged**